core/src/main/java/tc/oc/pgm/kits/ArmorKit.java

```diff
@@ -37,6 +39,11 @@ public void applyPostEvent(MatchPlayer player, boolean force, List<ItemStack> di
     for (Map.Entry<ArmorType, ArmorItem> entry : this.armor.entrySet()) {
       int slot = entry.getKey().ordinal();
       if (force || wearing[slot] == null || wearing[slot].getType() == Material.AIR) {
+        if (entry.getValue().teamColor) {
+          LeatherArmorMeta meta = (LeatherArmorMeta) entry.getValue().stack.getItemMeta();
+          meta.setColor(player.getParty().getFullColor());
+          entry.getValue().stack.setItemMeta(meta);
+        }
```

Comment:
This is modifying the item that should be immutable, apply this to the cloned stack instead (one line lower).
Also, i don't think you need to `setItemMeta` if you're getting the one and setting color

Reply:
I've tried it without setItemMeta and it gives me uncolored armor.

Reply:
> Also, i don't think you need to `setItemMeta` if you're getting the one and setting color

`getItemMeta` returns a clone and not a reference of the meta